### ISSUE TITLE
Adding model checkpoints to version control (policy + Git LFS / releases)

### ISSUE THREAD

**Issue by RohanPratapReddy** [veagy/DATA6950_Sentimental_Analysis_Cross_Domains#11]
# Add `checkpoints/` to version control (policy + Git LFS / releases)

## Summary

Define and implement a **clear policy** for what under **`checkpoints/`** belongs in Git versus **download scripts, Hugging Face cache, Git LFS, or release assets**. Trained **weights** (`.safetensors`, `.pt`, `.joblib`, large blobs) are usually **too large** for normal Git history; **metadata** (`run_meta.txt`, small `config.json` / tokenizer JSON) may be safe to commit if licenses allow.

## Motivation

- The capstone expects weights under paths such as **`checkpoints/2-labels/`**, **`checkpoints/3-labels/`**, **`checkpoints/pretrain/`**, plus **HF-style** folders under `checkpoints/deep_learning/llm/` (e.g. DistilBERT, BERT, RoBERTa, BART configs/tokenizers).
- Fresh clones **cannot run eval or resume training** without either committed weights or documented **restore** steps.
- Blind `git add checkpoints/` can **blow past** host limits (GitHub ~100 MB file warning, LFS quotas, slow clones).

## Recommended policy tiers (pick per artifact type)

| Tier | Examples | Suggested approach |
|------|----------|---------------------|
| **A — Small text / config** | `run_meta.txt`, `config.json`, `tokenizer_config.json` (no weights) | Commit to Git if redistributable |
| **B — Tokenizer / vocab JSON** | `tokenizer.json` (can be multi-MB) | Git LFS **or** “download on first run” from Hugging Face |
| **C — Fine-tuned / thesis weights** | `*.safetensors`, `*.joblib`, encoder checkpoints | **Git LFS** with explicit tracking **or** **GitHub Release** / S3 / Hugging Face Model Hub + manifest in repo |
| **D — Base model weights** | Full PyTorch/BERT binaries | **Do not commit**; use `transformers` cache + `snapshot_download` or documented env |

Document the chosen tier in **`docs/`** or root **`README.md`** and mirror it in **`output/models/checkpoints_inventory.csv`** export when available.

## Scope

**In scope**

- [ ] Audit **`checkpoints/`** size: `du -sh checkpoints/*` (or Windows equivalent); list largest files.
- [ ] Decide **minimum set** required for CI smoke (e.g. one tiny smoke checkpoint in `DUMMY/` vs real thesis weights).
- [ ] If committing binaries: enable **[Git LFS](https://git-lfs.github.com/)** and add `.gitattributes`, e.g.  
  `*.safetensors filter=lfs diff=lfs merge=lfs -text`  
  `*.joblib filter=lfs diff=lfs merge=lfs -text`  
  (add `.pt`, `.pth`, `.bin` if used).
- [ ] Add **`scripts/` or `docs/`** instructions: how to populate `checkpoints/` from releases/HF; checksums optional.
- [ ] Verify **licenses** for redistributing tokenizer dumps and fine-tuned weights derived from base models.
- [ ] Update **`.gitignore`** only if some subtrees must *never* be committed (e.g. local experiments); avoid ignoring paths needed for reproducibility.

**Out of scope**

- Training new checkpoints in this issue (only **storage and repo policy**).
- Paying for LFS bandwidth (team decision).

## Acceptance criteria

1. Written **policy** (README or `docs/checkpoints.md`) states what is committed vs downloaded.
2. If weights are in-repo: **LFS** (or releases) is configured; clone instructions include `git lfs pull` where needed.
3. Eval/train entrypoints match **documented paths** (or env vars) so a new machine can reproduce **one** end-to-end path (smoke or full).
4. PR merged; issue closed with link to storage choice (LFS vs Release vs Hub).

## Risks

- **Repository bloat** if someone commits large files without LFS (hard to purge from history).
- **Quota costs** on LFS or cloud egress.
- **Mismatch** between `Code/thesis/checkpoints/` (if any) and root `checkpoints/` — document both.

## Labels (suggestion)

`enhancement`, `area:checkpoints`, `needs-discussion`, `documentation`, `reproducibility`

## References

- Path: `checkpoints/` (repository root)
- Eval consumer: `Code/thesis/tools/eval_per_source_stem_metrics.py`
- Export: `Code/thesis/tools/export_model_documentation.py` → `output/models/checkpoints_inventory.csv`
- Related: `issues_markdown/issue-add-code-thesis-to-repo.md` (thesis tree may duplicate small HF caches)

---

**Comment by RohanPratapReddy** [comment]
https://github.com/veagy/DATA6950_Sentimental_Analysis_Cross_Domains/pull/18
